domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/stops/Entrance.java

```diff
@@ -0,0 +1,122 @@
+
+import java.util.Objects;
+
+import static org.mobilitydata.gtfsvalidator.domain.entity.stops.WheelchairBoarding.INHERIT_OR_UNKNOWN_WHEELCHAIR_BOARDING;
+
+/**
+ * Model class for an entity defined in stops.txt with location_type = 2
+ */
+public class Entrance extends LocationBase {
+    @NotNull
+    private final String parentStation;
+    @SuppressWarnings("NotNullFieldNotInitialized") //initialized through setter
+    @NotNull
+    private WheelchairBoarding wheelchairBoarding;
+
+    private Entrance(@NotNull String stopId,
+                     @Nullable String stopCode,
+                     @NotNull String stopName,
+                     @Nullable String stopDesc,
+                     @NotNull Float stopLat,
+                     @NotNull Float stopLon,
+                     @Nullable String zoneId,
+                     @Nullable String stopUrl,
+                     @NotNull String parentStation,
+                     @Nullable String stopTimezone,
+                     @Nullable WheelchairBoarding wheelchairBoarding) {
+        super(stopId, stopCode, stopName, stopDesc, stopLat, stopLon, zoneId, stopUrl, stopTimezone);
+        setWheelchairBoarding(wheelchairBoarding);
+        this.parentStation = parentStation;
+    }
+
+    public @NotNull String getParentStation() {
+        return parentStation;
+    }
+
+    public @NotNull WheelchairBoarding getWheelchairBoarding() {
+        return wheelchairBoarding;
+    }
+
+    public void setWheelchairBoarding(final WheelchairBoarding toSet) {
+        this.wheelchairBoarding = Objects.requireNonNullElse(toSet, INHERIT_OR_UNKNOWN_WHEELCHAIR_BOARDING);
+    }
+
+    public static class EntranceBuilder extends LocationBaseBuilder {
+
+        private String parentStation;
+        private WheelchairBoarding wheelchairBoarding = INHERIT_OR_UNKNOWN_WHEELCHAIR_BOARDING;
+        private Integer originalWheelchairBoarding = Integer.MAX_VALUE; // required to distinguish optional field not set
+
+        public EntranceBuilder parentStation(@NotNull final String parentStation) {
+            this.parentStation = parentStation;
+            return this;
+        }
+
+        public EntranceBuilder wheelchairBoarding(@Nullable final Integer wheelchairBoarding) {
+            this.wheelchairBoarding = WheelchairBoarding.fromInt(wheelchairBoarding);
+            this.originalWheelchairBoarding = wheelchairBoarding;
+            return this;
+        }
+
+        public EntityBuildResult<?> build() {
+            if (stopId == null || stopCode == null || stopLat == null || stopLon == null || parentStation == null) {
+                if (stopId == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("stops.txt", "stop_id", null));
+                }
+                if (stopCode == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("stops.txt", "stop_code", stopId));
```

Comment:
`stop_code` is optional, not required.

Could you add a process to have somebody (maybe Lionel) double checking those things?

Reply:
Yes, the process is having unit testing. Tests to come in subsequent commits before merge.

Reply:
https://github.com/MobilityData/gtfs-validator/blob/45601d0b30da2e58ccdb5760b9b0c9124bf70e1e/domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/stops/EntranceTest.java#L59